server/schedulers/balance_region.go

```diff
@@ -31,14 +32,19 @@ func init() {
 	})
 }
 
-// balanceRegionRetryLimit is the limit to retry schedule for selected store.
-const balanceRegionRetryLimit = 10
+const (
+	// balanceRegionRetryLimit is the limit to retry schedule for selected store.
+	balanceRegionRetryLimit  = 10
+	hintsStoreTTL            = time.Minute
+	hintsStoreCountThreshold = 1000
+)
 
 type balanceRegionScheduler struct {
 	*baseScheduler
 	selector     *schedule.BalanceSelector
 	taintStores  *cache.TTLUint64
 	opController *schedule.OperatorController
+	hintsCounter *hintsStoreBuilder
```

Comment:
Can we remove `taintStores` as `hitsFilter` seems to be a better solution for the same problem?